src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -303,7 +311,7 @@ private static function jsonEncode($value, int $flags = null, int $maxDepth = 51
             throw new InvalidArgumentException(sprintf('Invalid value for "json" option: %s.', $e->getMessage()));
         }
 
-        if (\PHP_VERSION_ID < 70300 && JSON_ERROR_NONE !== json_last_error() && (false === $value || !($flags & JSON_PARTIAL_OUTPUT_ON_ERROR))) {
+        if ((false === $value || !($flags & JSON_PARTIAL_OUTPUT_ON_ERROR)) && \PHP_VERSION_ID < 70300 && JSON_ERROR_NONE !== json_last_error()) {
```

Comment:
Same here, why is this way better?

Reply:
Same reason as in https://github.com/symfony/symfony/pull/31018#discussion_r273341842

Reply:
Actually this one is wrong if we use microoptimization arguments: `\PHP_VERSION_ID < 70300` is evaluated at compile time and putting it first enables dead code elimination.

Reply:
Reverted.